sentry/src/main/java/io/sentry/Scope.java

```diff
@@ -336,6 +337,42 @@ public void setContexts(final @NotNull String key, final @NotNull Object value)
     this.contexts.put(key, value);
   }
 
+  /**
+   * Sets the Scope's contexts
+   *
+   * @param key the context key
+   * @param value the context value
+   */
+  public void setContexts(final @NotNull String key, final @NotNull Boolean value) {
+    final Map<String, Boolean> map = new HashMap<>();
+    map.put("value", value);
```

Comment:
`h`: Why not just directly pass the value and key to `setContexts` and wrap it in a HashMap? We don't do this in [cocoa](https://github.com/getsentry/sentry-cocoa/blob/980d89d748992c1dcac6994eced3c074ff765917/Sources/Sentry/SentryScope.m#L149-L154) and [javascript](https://github.com/getsentry/sentry-javascript/blob/6c33bfcd27bb45cfbdc2459167c3b305ef93148c/packages/hub/src/scope.ts#L188-L198)

Reply:
@philipphofmann because your value is already a `NSDictionary`, set `setContexts` doesn't accept primitive types, but only Objects, otherwise the backend will complain.
so we wrap it for the common data types, but one is still able to pass a serializable object if they want to.
so if one does: setContext("a", "b")
it should be:

```
"a": {
value: "b"
}
```

and not:

`"a": "b"`

Reply:
if we change the main method setContexts(key, value (as a Map), people wont be able to do a setContexts(key, theirOwnInstanceOfASerializableClass)

Reply:
Ah, now I think I get your point. These are convenience methods, right? We can already do 
```
HashMap<String, Object> map = new HashMap<>();
map.put("name", "FireFighter");
map.put("age", 19);
Sentry.configureScope(scope -> scope.setContexts("character", map));
```
We don't have them on Cocoa. Why did you add them? Should I also add them in Cocoa?

Reply:
@philipphofmann it'd be nice to have those convenience methods, yes, but right now on Java it's kinda buggy.
`Contexts extends ConcurrentHashMap<String, Object>`

so one already could do setCotexts(x, y) but if `y` is a primitive type, it was not gonna be wrapped in an object like a `map` or so, so it'd trigger an error on the backend, every object on `contexts` should not be primitive, but an `object` but as in Java everything extends from an Object, the API allowed them to make a mistake.

Reply:
In Cocoa, we have `NSMutableDictionary<NSString *, NSDictionary<NSString *, id> *> *`, which allows you to put anything into the dict. We have
https://github.com/getsentry/sentry-cocoa/blob/master/Sources/Sentry/NSDictionary+SentrySanitize.m to make it work. Maybe something like this could help us also in Java?

Reply:
@philipphofmann this is already done automatically by Gson.
what I mean is, this still forces people to pass a Map, it's verbose IMO

Reply:
That is true indeed. Thanks for the explanation.